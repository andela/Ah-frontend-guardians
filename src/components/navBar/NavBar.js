import React , { Component } from 'react';
import store from '../../store/combineStore'
import {connect} from 'react-redux';
import avatar from '../../images/avatar.png';

export class NavBar extends Component{
    render(){
        return (
            <div className="nav">
                <nav className="navbar navbar-dark bg-dark" id="nav-bar">
                    <a href="#" id="logo">Author's Haven</a>
                    <span className="nav-links">
                        <span className="business-logic">
                            <a href="#">Home</a>
                            <a href="#">Articles</a>
                        </span>
                        <span>
                            {this.props.isLoggedIn ? <a id="avatar" href="#"><img src={avatar} className="avatar"></img></a>:
                                    <span className="auth-links">
                                        <a href="#">Signin</a>
                                        <a href="#">Register</a>
                                    </span>
                            }
                        </span>
                        
                    </span>
                </nav>
            </div>
        )
    }
}

const MapStateToProps = (state) => {
    return state.navBarReducer
}

export default connect(MapStateToProps)(NavBar);
