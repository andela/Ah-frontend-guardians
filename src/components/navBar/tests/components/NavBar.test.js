import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { NavBar} from '../../NavBar';
import avatar  from '../../src/images/avatar.png';

describe('Navbar Test', () => {

    const wrapper = shallow(<NavBar />)

    test('renders the component without crashing', () => {
        expect(wrapper).toMatchSnapshot()
    });


    test('renders the component', () => {
        expect(wrapper).toBeTruthy()
    })
})
