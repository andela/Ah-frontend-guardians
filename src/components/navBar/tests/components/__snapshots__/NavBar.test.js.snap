// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Test renders the component without crashing 1`] = `
<div
  className="nav"
>
  <nav
    className="navbar navbar-dark bg-dark"
    id="nav-bar"
  >
    <a
      href="#"
      id="logo"
    >
      Author's Haven
    </a>
    <span
      className="nav-links"
    >
      <span
        className="business-logic"
      >
        <a
          href="#"
        >
          Home
        </a>
        <a
          href="#"
        >
          Articles
        </a>
      </span>
      <span>
        <span
          className="auth-links"
        >
          <a
            href="#"
          >
            Signin
          </a>
          <a
            href="#"
          >
            Register
          </a>
        </span>
      </span>
    </span>
  </nav>
</div>
`;
